docs/docs/vm-fundamentals/builtins.md

```diff
@@ -1,7 +1,356 @@
+    
+    **`0 <= [p + 0] < 2^128`**
+    
+    **`0 <= [p + 1] < 2^128`**
+    
+    **`0 <= [p + 2] < 2^128`**
+    
+    Here, **`p`** represents the starting address of the built-in.
+    
+    To check if a value **`x`** falls within a smaller range **`[0,BOUND]`** where **`BOUND`** is less than **`2^128`**, two instances of the range-check are used:
+    
+    1. One instance confirms **`0 ≤ x < 2^128`**.
+    2. Another instance validates **`0 ≤ BOUND - x < 2^128`**.
+    
+    This approach facilitates computing integer division with a remainder using the range-check built-in. The objective is to calculate **`q = [x/y]`** and **`r = x mod y`**, rewritten as **`x = q*y + r`**, ensuring **`0 ≤ r < y`** holds true.
+    Care must be taken to prevent overflow during the **`x = q*y + r`** calculation. For simplicity, it's assumed that **`0 ≤ x, y < 2^64`**. Adjustments can be made to the code based on specific constraints if this assumption doesn't hold true.
+    
+    The provided code computes **`q`** and **`r`**, while ensuring **`0 ≤ x, y < 2^64`**, provided **`|F| > 2^128`**.
+    
+    ```rust
+    %builtins range_check
+    func div{range_check_ptr}(x, y) -> (q: felt, r: felt) {
+        alloc_locals;
+        local q;
+        local r;
+        %{ ids.q, ids.r = ids.x // ids.y, ids.x % ids.y %}
+    
+        // Check that 0 <= x < 2**64.
+        [range_check_ptr] = x;
+        assert [range_check_ptr + 1] = 2 ** 64 - 1 - x;
+    
+        // Check that 0 <= y < 2**64.
+        [range_check_ptr + 2] = y;
+        assert [range_check_ptr + 3] = 2 ** 64 - 1 - y;
+    
+        // Check that 0 <= q < 2**64.
+        [range_check_ptr + 4] = q;
+        assert [range_check_ptr + 5] = 2 ** 64 - 1 - q;
+    
+        // Check that 0 <= r < y.
+        [range_check_ptr + 6] = r;
+        assert [range_check_ptr + 7] = y - 1 - r;
+    
+        // Verify that x = q * y + r.
+        assert x = q * y + r;
+    
+        let range_check_ptr = range_check_ptr + 8;
+        return (q=q, r=r);
+    }
+    ```
+    
+- **ecdsa**
+    
+    A structure defines the memory layout for the signature builtin. It's employed by various functions in the common library, including the ecdsa builtin. For instance, in the verify_ecdsa_signature() function, there's an implicit argument of type SignatureBuiltin, which aids in tracking the next available builtin instance.
+    
+    The struct comprises two members of type felt:
+    
+    - **pub_key**: Represents an ECDSA public key.
+    - **message**: Denotes a message signed by the pub_key.
+    
+    Additionally, there's a pointer named ecdsa_ptr, which is of type SignatureBuiltin. This pointer is used to manage the signature builtin instances internally.
```

Comment:
which points to a variable of type `SignatureBuiltin`

or which is of type `SignatureBuiltin*`